build.py

```diff
@@ -2338,6 +2347,19 @@ def enable_all():
         required=False,
         help="Override specified backend CMake argument in the build as <backend>:<name>=<value>. The argument is passed to CMake as -D<name>=<value>. This flag only impacts CMake arguments that are used by build.py. To unconditionally add a CMake argument to the backend build use --extra-backend-cmake-arg.",
     )
+    parser.add_argument(
+        "--ci-split",
```

Comment:
If it would have any value to end users as well, how about naming it something like `--split-build` or something instead, not specific to CI?

Reply:
it should be work only in CI we shouldn't expose it to the customers.
You have a valid point `ci` vs `build` where second sound more reasonable. 
I have no expectation that users will use it.

Reply:
PR LGTM assuming CI build is passing using these new flags. Would still prefer to rename this flag to mention `build` instead of `ci`.